### PR TITLE
production CD with self-hosted github runner

### PR DESCRIPTION
**Pull request by 123joshuawu**
I've setup a self-hosted Github Action runner on the production VM so we can finally do CD through Github Actions.

The `prod-cd` workflow runs on master push or manual trigger. It uses the `yacs-prod` environment which contains the secrets for prod as well as configuration for manual verification of deployment. I've currently configured it such that deployments require manual approval (by me for now).

The `prod-cd` workflow will pull the new code from master, rebuild and restart `yacs_api` and `yacs_web` in the correct order. 

I've done some quick tests, you can go back in the commits or actions history to see the workflow in action. 

TLDR; CD process = push to master -> create new deployment -> manual approve -> redeploy production

### CODE REVIEW

File: .github/workflows/prod-cd.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    runs-on: [self-hosted, rpi, prod]
+    environment: yacs-prod
```
Comment (jshom) [inline .github/workflows/prod-cd.yaml:12]: lets rename this to `yacs-rpi-prod` since rpi specific.
besides that, awesome!

**Review by jshom — APPROVED** [review]

File: .github/workflows/prod-cd.yaml (hunk 1 above)
Comment (123joshuawu) [reply .github/workflows/prod-cd.yaml]: Yep changed
